src/Settings.h

```diff
@@ -29,24 +34,24 @@ namespace weathersvr {
 
         private:
 
-            void setDefaultValues();
+            //If there is a problem opening file, parsing or Dark Sky API key seems to be invalid - set the flag to terminate the program.
+            void processSettingsFile(std::string& currentDir);
+
             /*
             Check if the values present in the settings.json file related to the dark sky api are valid before set them to respective variables. If is not valid, the default values will be kept.
             This function may throw an exception web::json::json_exception.
-            @param web::json::value& jsonObj - The Json OBJECT which the values will be parsed and validated.
+            @param web::json::value& jsonObj - The Json OBJECT which values will be parsed and validated.
             */
-            void validateValuesFromDarkSky(web::json::value& jsonObj);
+            bool validateValuesFromDarkSky(web::json::value& jsonObj);
 
             utility::string_t keyApiDarksky;
             utility::string_t units;
-            short intervalDownloadWeatherData;
-
-            //was public, changed to private
+            int intervalWeatherDataDownload;
             int port_number;
             std::string endpointUrl;
             std::string hostName;
 
-            //Flag to track if we get invalid DarkSky API key from settings file. Can stop sending weather requests etc
-            bool validkeyApiDarksky = true;
+            //using this flag to check if we need to terminate the program
+            bool settingsAreValid;
```

Comment:
Comment is a bit strange. Its up to client code to decide if it will terminate the program or not.
Also we can initialize the value here to `false`.

Reply:
Fixed. Just wasn't sure about the practice initializing variables in header instead of .c file.